client/session.go

```diff
@@ -702,6 +713,94 @@ func (s *session) Truncate(namespace string) (int64, error) {
 	return truncated, resultErr.FinalError()
 }
 
+func (s *session) getPeersForShard(shard uint32) ([]hostQueue, error) {
+	s.RLock()
+	peers := make([]hostQueue, 0, s.topoMap.Replicas())
+	err := s.topoMap.RouteShardForEach(shard, func(idx int, host topology.Host) {
+		if s.origin != nil && s.origin.ID() == host.ID() {
+			// Don't include the origin host
+			return
+		}
+		peers = append(peers, s.queues[idx])
+	})
+	s.RUnlock()
+	if err != nil {
+		return nil, err
+	}
+	return peers, nil
+}
+
+type metadataIter struct {
```

Comment:
Can just move to bottom of file then perhaps?  Main issue is just being sandwiched between `*session` methods.